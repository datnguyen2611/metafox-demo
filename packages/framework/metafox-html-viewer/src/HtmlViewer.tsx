import React from 'react';
import ReactHtmlParser from 'react-html-parser';
import default_transform from './transform';
import simple_transform from './simpleTransform';
import { HtmlComponentProps } from './types';

function nl2br(text: string): string {
  // issue migrate from v4. </br> make bad layout
  return `${text}`.replace(/([^>\r\n]?)(\r\n|\n\r|\r|\n)/g, '$1<p></p>');
}

const HtmlComponent = ({
  html,
  decodeEntities = true,
  transform: transformProp,
  component: Wrap = React.Fragment,
  disableNl2br,
  preprocessNodes,
  simpleTransform = false
}: HtmlComponentProps) => {
  const transform =
    transformProp || (simpleTransform ? simple_transform : default_transform);

  return (
    <Wrap>
      {html
        ? ReactHtmlParser(disableNl2br ? html : nl2br(html), {
            decodeEntities,
            transform,
            preprocessNodes
          })
        : null}
    </Wrap>
  );
};

export default HtmlComponent;
