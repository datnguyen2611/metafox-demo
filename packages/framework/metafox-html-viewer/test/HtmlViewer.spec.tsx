import HtmlViewer from '@metafox/html-viewer';
import { render, waitFor } from '@testing-library/react';
import React from 'react';
import { MemoryRouter } from 'react-router';

/* eslint-disable no-multi-str */
const html =
  "Leading news sites in U.S. Europe Asia.\
   <a data-testid='myLink' href='/jorge21' target='_blank'>Jade McCullough V</a> \
  \nSearch and Browse archive.. See Early Warning. \n\nQuality sites in U.S. Europe, \
  Asia. See Renewal America, Renewal Europe. \nSearch and browse";

/* eslint-disable no-multi-str*/
describe('HtmlViewer', () => {
  it('+render', async () => {
    const { asFragment, queryByTestId } = render(
      <MemoryRouter>
        <HtmlViewer
          html={html}
          component={({ children }) => (
            <div data-testid="myComponent">{children}</div>
          )}
        />
      </MemoryRouter>
    );

    await waitFor(() => expect(queryByTestId('myComponent')));

    expect(asFragment()).toMatchSnapshot();
  });
});
