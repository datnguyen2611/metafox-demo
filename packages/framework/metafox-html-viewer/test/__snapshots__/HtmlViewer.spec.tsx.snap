// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HtmlViewer +render 1`] = `<DocumentFragment />`;
